### ISSUE TITLE
StringIndexOutOfBoundsException causes crash

### ISSUE THREAD

**Issue by ubaierbhat** [elastic/apm-agent-android#242]
StringIndexOutOfBoundsException on some devices.  Please check `telephonyManager.getSimOperator()` is returing a non null String of expected lenght. 

version: `0.11.0`

```
Caused by java.lang.StringIndexOutOfBoundsException: length=0; index=3
       at java.lang.String.substring(String.java:2060)
       at co.elastic.apm.android.sdk.internal.services.network.NetworkService.getCarrierInfo(NetworkService.java:82)
       at co.elastic.apm.android.sdk.attributes.common.CarrierHttpAttributesVisitor.visit(CarrierHttpAttributesVisitor.java:39)
       at co.elastic.apm.android.sdk.attributes.impl.ComposeAttributesVisitor.visit(ComposeAttributesVisitor.java:36)
       at co.elastic.apm.android.sdk.attributes.impl.VisitorAttributesCreator.create(VisitorAttributesCreator.java:36)
       at co.elastic.apm.android.sdk.internal.opentelemetry.processors.spans.ElasticSpanProcessor.onStart(ElasticSpanProcessor.java:51)
       at io.opentelemetry.sdk.trace.SdkSpan.startSpan(SdkSpan.java:197)
       at io.opentelemetry.sdk.trace.SdkSpanBuilder.startSpan(SdkSpanBuilder.java:219)
       at co.elastic.apm.android.sdk.traces.http.impl.okhttp.OtelOkHttpEventListener.createWrapperSpan(OtelOkHttpEventListener.java:111)
       at co.elastic.apm.android.sdk.traces.http.impl.okhttp.OtelOkHttpEventListener.callStart(OtelOkHttpEventListener.java:84)
       at java.lang.reflect.Method.invoke(Method.java)
       at co.elastic.apm.android.common.okhttp.eventlistener.CompositeEventListener.doCall(CompositeEventListener.java:41)
       at co.elastic.apm.android.common.okhttp.eventlistener.Generated_CompositeEventListener.callStart(:20)
       at okhttp3.internal.connection.RealCall.callStart(RealCall.kt:171)
       at okhttp3.internal.connection.RealCall.enqueue(RealCall.kt:163)
       at okhttp3.internal.ws.RealWebSocket.connect(RealWebSocket.kt:165)
       at okhttp3.OkHttpClient.newWebSocket(OkHttpClient.kt:281)
       at io.socket.engineio.client.transports.WebSocket.doOpen(WebSocket.java:52)
       at io.socket.engineio.client.Transport$1.run(Transport.java:76)
       at io.socket.thread.EventThread.exec(EventThread.java:55)
       at io.socket.engineio.client.Transport.open(Transport.java:71)
       at io.socket.engineio.client.Socket$2.run(Socket.java:264)
       at io.socket.thread.EventThread.exec(EventThread.java:55)
       at io.socket.engineio.client.Socket.open(Socket.java:242)
       at io.socket.client.Manager$1.run(Manager.java:305)
       at io.socket.thread.EventThread.exec(EventThread.java:55)
       at io.socket.client.Manager.open(Manager.java:222)
       at io.socket.client.Manager$7$1.run(Manager.java:517)
       at io.socket.thread.EventThread$2.run(EventThread.java:80)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

**Comment by LikeTheSalad** [comment]
Thanks for your feedback, @ubaierbhat - I'll take a look!

**Comment by LikeTheSalad** [comment]
The fix for this issue is merged and will be available in the next release `0.13.0`.
